src/Tasks/Microsoft.NET.Build.Tasks/FileGroup.cs

```diff
@@ -83,7 +83,7 @@ public static string GetTypeMetadata(this FileGroup fileGroup)
                     return package.FrameworkAssemblies.Select(c => Tuple.Create(c, _emptyProperties));
 
                 default:
-                    throw new Exception($"Unexpected file group in project.lock.json target library {package.Name}");
+                    throw new BuildErrorException(Strings.UnexpectedFileGroupInAssetsFile, package.Name);
```

Comment:
So, the caller of this code is doing it for each value in the enum. If we get here, we have a value outside the enum range and ToString() won't help much. This is indeed a bug to get here, so I'll just throw ArgumentOutOfRangeException(nameof(fileGroup)).

---
In reply to: [88394333](https://github.com/dotnet/sdk/pull/398#discussion_r88394333) [](ancestors = 88394333)